src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/ApiSteps.java

```diff
@@ -182,12 +180,17 @@ public void cleanUp() {
             }
         }
 
-        for (Jurisdiction jurisdiction : testContext.getScenarioContext().getJurisdictions()) {
-            try {
-                jurisdictionRepository.delete(jurisdiction);
-            } catch (DataIntegrityViolationException e) {
-                log.error(
-                    "Failure may be due to foreign key. This is okay because the online hearing will be deleted elsewhere.");
+        List<Jurisdiction> jurisdictions = testContext.getScenarioContext().getJurisdictions();
+        if (jurisdictions != null && !jurisdictions.isEmpty()) {
+            for (Jurisdiction jurisdiction : jurisdictions) {
+                try {
```

Comment:
The session event register depends on Jurisdiction. So remove those first if there are any test registers linked to the the jurisdiction